ArcGISRuntimeSDKQt_CppSamples/Maps/GenerateOfflineMap/GenerateOfflineMap.cpp

```diff
@@ -103,9 +104,8 @@ void GenerateOfflineMap::generateMapByExtent(double xCorner1, double yCorner1, d
   const Envelope extent = Envelope(corner1, corner2);
   const Envelope mapExtent = geometry_cast<Envelope>(GeometryEngine::project(extent, SpatialReference::webMercator()));
 
-  // connect to the signal for when the default parameters are generated
-  connect(m_offlineMapTask, &OfflineMapTask::createDefaultGenerateOfflineMapParametersCompleted,
-          this, [this](const QUuid&, const GenerateOfflineMapParameters& params)
+  // generate parameters
+  m_offlineMapTask->createDefaultGenerateOfflineMapParametersAsync(mapExtent).then(this, [this](GenerateOfflineMapParameters params)
```

Comment:
```suggestion
  m_offlineMapTask->createDefaultGenerateOfflineMapParametersAsync(mapExtent).then(this, [this](const GenerateOfflineMapParameters& params)
```

There are probably others. They should all take args by const-ref. I won't comment on them all.